src/vendors/autocomplete.js

```diff
@@ -1,140 +1,108 @@
-    that = elems[i];
-    that.last_val = '';
-
-    const cancelObsolete = function() {
-      clearTimeout(that.timer);
-      if (that.sourcePending) {
-        that.sourcePending.abort();
-        that.sourcePending = null;
-      }
-    };
+    if (value.length < this.minChars) {
+      return this.previousValue = value
+    }
 
-    const suggest = function(data) {
-      cancelObsolete();
-      const val = that.value;
-      if (data && val.length >= o.minChars) {
-        o.renderItems(data, val);
-      }
-    };
-
-
-    that.inputHandler = function() {
-      const val = that.value;
-      if (val.length >= o.minChars) {
-        if (val != that.last_val) {
-          cancelObsolete();
-          that.last_val = val;
-          that.timer = setTimeout(function() {
-            // @HACK: a bug in Firefox for Android (https://bugzilla.mozilla.org/show_bug.cgi?id=1610083)
-            // triggers a redundant 'input' event on the field just before it's blurred,
-            // resulting in the suggest list re-appearing after a suggestion has been made.
-            // So we check if the element having the focus is the field before doing anything.
-            if (document.activeElement && document.activeElement !== that) {
-              return;
-            }
-            that.sourcePending = o.source(val);
-            that.sourcePending.then(source => {
-              that.sourcePending = null;
-              if (source !== null && document.activeElement === that) {
-                suggest(source);
-              }
-            }).catch(e => {
-              console.warn(e); /* should be handled by a telemetry logger */
-              that.sourcePending = null;
-            });
-          }, o.delay);
-        }
-      } else {
-        that.last_val = val;
+
+    this.cancelObsolete();
+    this.previousValue = value;
+    this.timer = setTimeout(() => {
+      // @HACK: a bug in Firefox for Android (https://bugzilla.mozilla.org/show_bug.cgi?id=1610083)
+      // triggers a redundant 'input' event on the field just before it's blurred,
+      // resulting in the suggest list re-appearing after a suggestion has been made.
+      // So we check if the element having the focus is the field before doing anything.
+      if (document.activeElement && document.activeElement !== this.input) {
+        return;
       }
-    };
-    addEvent(that, 'input', that.inputHandler);
-
-    that.focusHandler = function(e) {
-      that.last_val = '\n';
-      that.inputHandler(e);
-    };
-    if (!o.minChars) {
-      addEvent(that, 'focus', that.focusHandler);
+
+      this.sourcePending = this.source(value);
+      this.sourcePending
+        .then((source) => {
+          if (source !== null && document.activeElement === this.input) {
+            this.suggest(source);
+          }
+        })
+        .catch((e) => {
+          console.warn(e); /* should be handled by a telemetry logger */
+        });
+
+      this.sourcePending = null;
```

Comment:
:warning: This is probably not equivalent to the previous behavior. 
`sourcePending` should not be set to `null` before the promise resolved (and the HTTP response has been received).